### PR TITLE
Add a span for OpenTelemetry to Prometheus push request conversion in distributors

### PR DESCRIPTION
**Pull request by charleskorn**
#### What this PR does

This PR improves the observability of OTLP write requests in distributors.

We've seen instances where the conversion to Prometheus write request format can take up to a second (which inspired #5531). This PR adds a span with more information so we can more easily diagnose cases like these in the future.

#### Which issue(s) this PR fixes or relates to

Relates to #5531 

#### Checklist

- [n/a] Tests updated
- [n/a] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/push/otel.go [hunk 1]
```diff
@@ -109,16 +111,44 @@ func OTLPHandler(
...
+		log.SetTag("content_type", contentType)
+		log.SetTag("content_encoding", contentEncoding)
+		log.SetTag("content_length", r.ContentLength)
+
 		otlpReq, err := decoderFunc(body)
 		if err != nil {
 			return body, err
 		}
 
+		level.Debug(log).Log("msg", "decoding complete")
```
Comment (pstibrany) [inline pkg/util/push/otel.go:126]: nit: improve message to indicate that we're not done yet despite "complete" in the message.

```suggestion
		level.Debug(log).Log("msg", "decoding complete, starting conversion")
```

**Review by pstibrany — APPROVED** [review]
thanks!

File: pkg/util/push/otel.go [hunk 2]
```diff
@@ -109,16 +111,44 @@ func OTLPHandler(
 			return body, err
 		}
 
+		log, ctx := spanlogger.New(ctx, "Distributor.OTLPHandler.decodeAndConvert")
```
Comment (56quarters) [inline pkg/util/push/otel.go:114]: Probably want `spanlogger.NewWithLogger` and use the existing logger from above. Though, the logger from above is still the global logger which isn't great.

**Review by 56quarters — APPROVED** [review]

File: pkg/util/push/otel.go (hunk 2 above)
Comment (charleskorn) [reply pkg/util/push/otel.go]: Good point, fixed in 42eaf65

**Pull request merged**